Construction of default pass with optimization level 3 fails

The error is:
```
>       return SequencePass(passlist)
E       RuntimeError: Cannot compose these Compiler Passes due to mismatching Predicates of type: GateSetPredicate
```

This shows up in [this test](https://github.com/CQCL/pytket-quantinuum/blob/07a436709338c47d092f6613e82eb6351e2a5911/tests/integration/backend_test.py#L1542). See for example [this workflow run](https://github.com/CQCL/pytket-quantinuum/actions/runs/13592306919/job/38001128412).

--- a/_metadata.py
+++ b/_metadata.py
@@ -1,2 +1,2 @@
-__extension_version__ = "0.44.0"
+__extension_version__ = "0.45.0"
 __extension_name__ = "pytket-quantinuum"

--- a/docs/changelog.md
+++ b/docs/changelog.md
@@ -4,10 +4,14 @@
 
 # Changelog
 
-## Unreleased
+## 0.45.0 (March 2025)
 
 - Add CliffordCircuitPredicate for circuit compilation and processing
   when using the stabilizer simulator.
+- Update pytket version requirement to 2.0.1.
+- Update pytket-qir version requirement to 0.21.
+- Update pytket-pecos version requirement to 0.2.0.
+
 
 ## 0.44.0 (February 2025)
 

--- a/pytket/extensions/quantinuum/backends/quantinuum.py
+++ b/pytket/extensions/quantinuum/backends/quantinuum.py
@@ -116,7 +116,6 @@ _ADDITIONAL_GATES = {
     OpType.ExplicitModifier,
     OpType.SetBits,
     OpType.CopyBits,
-    OpType.ClassicalExpBox,
     OpType.ClExpr,
     OpType.WASM,
 }

--- a/setup.py
+++ b/setup.py
@@ -45,7 +45,7 @@ setup(
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket >= 2.0.0",
+        "pytket >= 2.0.1",
         "pytket-qir >= 0.21, < 0.22",
         "requests >= 2.32.2",
         "types-requests",

--- a/setup.py
+++ b/setup.py
@@ -45,8 +45,8 @@ setup(
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket >= 1.40.0, < 2",
-        "pytket-qir >= 0.20, < 0.21",
+        "pytket >= 2.0.0",
+        "pytket-qir >= 0.21, < 0.22",
         "requests >= 2.32.2",
         "types-requests",
         "websockets >= 13.1",
@@ -56,7 +56,7 @@ setup(
         "numpy >= 1.26.4",
     ],
     extras_require={
-        "pecos": ["pytket-pecos ~= 0.1.32"],
+        "pecos": ["pytket-pecos ~= 0.2.0"],
         "calendar": ["matplotlib >= 3.8.3,< 3.11.0", "pandas ~= 2.2.1"],
     },
     classifiers=[

--- a/tests/integration/backend_test.py
+++ b/tests/integration/backend_test.py
@@ -46,6 +46,7 @@ from pytket.circuit import (
     reg_lt,
     reg_neq,
 )
+from pytket.circuit.clexpr import wired_clexpr_from_logic_exp
 from pytket.extensions.quantinuum import (
     Language,
     QuantinuumBackend,
@@ -412,10 +413,10 @@ def test_classical(
     c.add_c_setreg(23, a)
     c.add_c_copyreg(a, b)
 
-    c.add_classicalexpbox_register(a + b, d.to_list())
-    c.add_classicalexpbox_register(a - b, d.to_list())
-    c.add_classicalexpbox_register(a << 1, a.to_list())
-    c.add_classicalexpbox_register(a >> 1, b.to_list())
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a + b, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a - b, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a << 1, a.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a >> 1, b.to_list()))
 
     c.X(0, condition=reg_eq(a ^ b, 1))
     c.X(0, condition=(a[0] ^ b[0]))
@@ -465,7 +466,7 @@ def test_division(
 
     c.add_c_setbits([False, True] + [False] * 6, a)  # type: ignore
     c.add_c_setbits([True, True] + [False] * 8, b)  # type: ignore
-    c.add_classicalexpbox_register(a * b // d, d.to_list())
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a * b // d, d.to_list()))
 
     backend = authenticated_quum_backend_qa
 

--- a/tests/integration/backend_test.py
+++ b/tests/integration/backend_test.py
@@ -432,6 +432,8 @@ def test_classical(
     c.X(0, condition=reg_leq(a, 1))
     c.Phase(0, condition=a[0])
 
+    c.measure_all()
+
     backend = authenticated_quum_backend_qa
 
     c = backend.get_compiled_circuit(c)
@@ -793,6 +795,7 @@ def test_wasm_qa(
     c.name = "test_wasm"
     a = c.add_c_register("a", 8)
     c.add_wasm_to_reg("add_one", wasfile, [a], [a])
+    c.measure_all()
 
     b = authenticated_quum_backend_qa
 
@@ -862,6 +865,7 @@ def test_wasm_costs(
     c.name = "test_wasm"
     a = c.add_c_register("a", 8)
     c.add_wasm_to_reg("add_one", wasfile, [a], [a])
+    c.measure_all()
 
     b = authenticated_quum_backend_qa
 

--- a/tests/integration/local_emulator_test.py
+++ b/tests/integration/local_emulator_test.py
@@ -31,6 +31,7 @@ from pytket.circuit import (
     reg_lt,
     reg_neq,
 )
+from pytket.circuit.clexpr import wired_clexpr_from_logic_exp
 from pytket.extensions.quantinuum import QuantinuumBackend, have_pecos
 from pytket.wasm import WasmFileHandler
 
@@ -169,11 +170,11 @@ def test_classical_0(authenticated_quum_backend_prod: QuantinuumBackend) -> None
     c.add_c_setreg(23, a)
     c.add_c_copyreg(a, b)
 
-    c.add_classicalexpbox_register(a + b, d.to_list())
-    c.add_classicalexpbox_register(a - b, d.to_list())
-    c.add_classicalexpbox_register(a * b * d, d.to_list())
-    c.add_classicalexpbox_register(a << 1, a.to_list())
-    c.add_classicalexpbox_register(a >> 1, b.to_list())
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a + b, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a - b, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a * b * d, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a << 1, a.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a >> 1, b.to_list()))
 
     c.X(0, condition=reg_eq(a ^ b, 1))
     c.X(0, condition=reg_eq(a & b, 1))
@@ -216,11 +217,11 @@ def test_classical_1(authenticated_quum_backend_prod: QuantinuumBackend) -> None
     c.add_c_setreg(23, a)
     c.add_c_copyreg(a, b)
 
-    c.add_classicalexpbox_register(a + b, d.to_list())
-    c.add_classicalexpbox_register(a - b, d.to_list())
-    c.add_classicalexpbox_register(a * b * d, d.to_list())
-    c.add_classicalexpbox_register(a << 1, a.to_list())
-    c.add_classicalexpbox_register(a >> 1, b.to_list())
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a + b, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a - b, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a * b * d, d.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a << 1, a.to_list()))
+    c.add_clexpr(*wired_clexpr_from_logic_exp(a >> 1, b.to_list()))
 
     c.X(0, condition=reg_eq(a ^ b, 1))
     c.Measure(Qubit(0), d[0])
@@ -245,7 +246,7 @@ def test_classical_2(authenticated_quum_backend_prod: QuantinuumBackend) -> None
     b = circ.add_c_register("b", 2)
     c = circ.add_c_register("c", 1)
     expr = a[0] ^ b[0]
-    circ.add_classicalexpbox_bit(expr, [c[0]])
+    circ.add_clexpr(*wired_clexpr_from_logic_exp(expr, [c[0]]))
     circ.X(0)
     circ.Measure(Qubit(0), a[1])
     backend = authenticated_quum_backend_prod
@@ -270,8 +271,8 @@ def test_classical_3(authenticated_quum_backend_prod: QuantinuumBackend) -> None
     circ.add_c_setreg(3, a)
     circ.add_c_copyreg(a, b)
 
-    circ.add_classicalexpbox_register(a - b, c.to_list())
-    circ.add_classicalexpbox_register(a << 1, a.to_list())
+    circ.add_clexpr(*wired_clexpr_from_logic_exp(a - b, c.to_list()))
+    circ.add_clexpr(*wired_clexpr_from_logic_exp(a << 1, a.to_list()))
 
     circ.X(0)
     circ.Measure(Qubit(0), a[3])

